docs/concepts/caching.rst

```diff
@@ -0,0 +1,489 @@
+
+    @tag(caching_behavior="dont_fingerprint")
+    def weather_predictor(
+        current_temperature: float, current_wind: dict, is_rain: bool,
+    ) -> str:
+        """Train a model to predict the probability of rain"""
+        model = HistGradientBoostingRegressor(...)
+        model.train(...)
+        return model
+
+.. note::
+
+     An alternative solution is to register a custom fingerprinting function by ``type`` (see **Fingerprinting**).
+
+
+Don't store result
+~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+   :align: center
+
+   * - Try to retrieve result
+     - Store result
+     - Fingerprint result
+     - Store fingerprint
+   * - ✅
+     - ❌
+     - ✅
+     - ❌
+
+If a node produces results that are large in size, not picklable, or very inexpensive to compute, it might be preferrable to not store it. Also, your environment may have storage or I/O limitations (e.g., web service).
+
+
+Specifying ``caching_behavior="dont_store_result"`` means Hamilton won't add the result to the result store, but the fingerprint won't be added to the metadata store since it would point to nothing (i.e., "result store miss"). The fingerprint is only used to create the key for children nodes (e.g., ``recommended_outfit``).
+
+For instance, we don't want to store the ``weather_message`` because it's only a set of ``if/else`` that are trivial to compute.
+
+.. code-block:: python
+
+    import llm
+    import weather_api
+
+    def weather_client(credentials: dict) -> weather_api.Client:
+        return weather_api.Client(**credentials)
+
+    def current_temperature(weather_client: weather_api.Client) -> float:
+        """Get the current local temperature from a public API"""
+        return weather_api.Client(**credentials).get_temperature(...)
+
+    @tag(caching_behavior="dont_store_result")
+    def weather_message(
+        current_temperature: float,
+        current_wind: dict,
+        is_rain: bool,
+    ) -> str:
+        """Interpret the temperature and return a message"""
+        if ...:
+            return "what a nice day 🌞"
+        elif ...:
+            return "you should bring your umbrella ☔"
+        else:
+            return "it's starting to feel cold 🥶"
+
+    def recommend_outfit(ll_client: llm.Client, weather_message: str) -> str:
+        """Use an LLM to generate an outfit suggestion based on the weather"""
+        prompt = f"Make an outfit suggestion based on: {weather_message}""
+        response = llm.Client.generate(prompt)
+        return response
+
+
+.. note::
+
+    Even when ``caching_behavior="dont_store_result"``, the adapter will try to retrieve stored values that could have been produce before this behavior was set.
+
+
+Summary
+~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+   :align: center
+
+   * - Behavior
+     - Try to retrieve result
+     - Store result
+     - Fingerprint result
+     - Store fingerprint
```

Comment:
When reading this table I don't know what I need to do.
This should be `@tag(...)` since that's what you want people to do, right?

Reply:
This is the summary table from other subheaders

Reply:
I understand -- but I need to scroll up to make that connection.